File: Facade/Dart/NonVegRestaurant.dart
import 'IRestaurant.dart';
import 'Menu.dart';

class NonVegRestaurant implements IRestaurant {
  Menu? _menu;

  NonVegRestaurant() {
    _menu = Menu("Non Veg Menu");
  }

  @override
  Menu? getMenus() {
    return _menu;
  }
}

File: Facade/Javascript/NonVegRestaurant.js
import IRestaurant from "./IRestarutant.js";
import Menu from "./Menu.js";

class NonVegRestaurant extends IRestaurant {
  constructor() {
    super();
    this.menu = new Menu("Non Veg Menu");
  }

  getMenus() {
    return this.menu;
  }
}

export default NonVegRestaurant;